homeassistant/components/camera/proxy.py

```diff
@@ -0,0 +1,264 @@
+    image = yield from stream.read(length)
+    return image
+
+
+def _resize_image(image, opts):
+    """Resize image."""
+    from PIL import Image
+    import io
+
+    if not opts:
+        return image
+
+    quality = opts.quality if opts.quality else DEFAULT_QUALITY
+    new_width = opts.max_width
+
+    img = Image.open(io.BytesIO(image))
+    imgfmt = str(img.format)
+    if imgfmt != 'PNG' and imgfmt != 'JPEG':
+        _LOGGER.debug("Image is of unsupported type: %s", imgfmt)
+        return image
+
+    (old_width, old_height) = img.size
+    old_size = len(image)
+    if old_width <= new_width:
+        if opts.quality is None:
+            _LOGGER.debug("Image is smaller than requested width")
+            return image
+        new_width = old_width
+
+    scale = new_width / float(old_width)
+    new_height = int((float(old_height)*float(scale)))
+
+    img = img.resize((new_width, new_height), Image.ANTIALIAS)
+    imgbuf = io.BytesIO()
+    img.save(imgbuf, "JPEG", optimize=True, quality=quality)
+    newimage = imgbuf.getvalue()
+    if not opts.force_resize and len(newimage) >= old_size:
+        _LOGGER.debug("Using original image(%d bytes) "
+                      "because resized image (%d bytes) is larger",
+                      old_size, len(newimage))
+        return image
+
+    _LOGGER.debug("Resized image "
+                  "from (%dx%d - %d bytes) "
+                  "to (%dx%d - %d bytes)",
+                  old_width, old_height, old_size,
+                  new_width, new_height, len(newimage))
+    return newimage
+
+
+class ImageOpts():
+    """The representation of image options."""
+
+    def __init__(self, max_width, quality, force_resize):
+        """Initialize image options."""
+        self.max_width = max_width
+        self.quality = quality
+        self.force_resize = force_resize
+
+    def __bool__(self):
+        """Bool evaution rules."""
+        return True if self.max_width or self.quality else False
+
+
+class ProxyCamera(Camera):
+    """The representation of a Proxy camera."""
+
+    def __init__(self, hass, config):
+        """Initialize a proxy camera component."""
+        super().__init__()
+        self.hass = hass
+        self._proxied_camera = config.get(CONF_ENTITY_ID)
+        self._name = config.get(CONF_NAME)
+        if not self._name:
+            self._name = "{} - {}".format(DEFAULT_BASENAME,
+                                          self._proxied_camera)
+        self._image_opts = ImageOpts(
+            config.get(CONF_MAX_IMAGE_WIDTH),
+            config.get(CONF_IMAGE_QUALITY),
+            config.get(CONF_FORCE_RESIZE))
+
+        self._stream_opts = ImageOpts(
+            config.get(CONF_MAX_STREAM_WIDTH),
+            config.get(CONF_STREAM_QUALITY),
+            True)
+        self._image_refresh_rate = config.get(CONF_IMAGE_REFRESH_RATE)
+        self._cache_images = (
+            True
+            if (config.get(CONF_IMAGE_REFRESH_RATE)
```

Comment:
why does CONF_IMAGE_REFRESG_RATE trigger cache ?

Reply:
Because the frontend makes a request at an arbitrary rate so you must cache the results when the refresh rate is specified (strictly speaking only when slower than the frontend rate, but there is no way to know what that is)

Reply:
Multilines if statements are unreadable. Also, in this case you just want True or False, so just cast the if-condition to a bool.